tool/tsh/common/tsh.go

```diff
@@ -3225,17 +3247,54 @@ func accessRequestForSSH(ctx context.Context, _ *CLIConf, tc *client.TeleportCli
 
 	// At this point we have exactly 1 node.
 	node := rsp.Servers[0]
+	var req types.AccessRequest
 	requestResourceIDs := []types.ResourceID{{
 		ClusterName: tc.SiteName,
 		Kind:        types.KindNode,
 		Name:        node.GetName(),
```

Comment:
Hmm... we're getting this node above via `GetSSHTargets` which should only return resources that the user already has access to or could see via `SearchAsRoles`, does this work if the user does not have any SearchAsRoles?

Reply:
No, and I think that's okay. I don't think a user would typically try to make a request in `tsh ssh` if they didn't know what resource they were SSHing into, and if they wanted to request a role without knowing any resources they'd use `tsh requests create`.

Reply:
So are we okay with requiring the user to have both `search_as_roles` and `roles` set in order to do an automatic role request?

If we are, that might actually be a solution to the read access issue forrest brought up. We would need to make sure that `GetAccessCapabilities` wouldn't leak any info about resource existence unless the user can already see all those resources via their base role or search_as_roles